blobstore/presigned_url.go

```diff
@@ -188,34 +189,68 @@ func (bh *BlobHandler) HandleGetPresignedURLMultiObj(c echo.Context) error {
 		return c.JSON(http.StatusNotFound, errMsg.Error())
 	}
 	//check if size is below 5GB
-	size, fileCount, err := bh.GetSize(response)
+	size, _, err := bh.GetSize(response)
 	if err != nil {
-		log.Error("HandleGetPresignedURLMultiObj: Error getting size:", err.Error())
-		return c.JSON(http.StatusInternalServerError, err.Error())
+		errMsg := fmt.Errorf("error getting size: %s", err.Error())
+		log.Error(errMsg.Error())
+		return c.JSON(http.StatusInternalServerError, errMsg.Error())
+	}
+	var downloadLimit int
+	downloadLimit, err = strconv.Atoi(os.Getenv("ZIP_DOWNLOAD_SIZE_LIMIT"))
+	if err != nil {
+		log.Debugf("size download limit defaulted to %v", DEFAULT_ZIP_DOWNLOAD_SIZE_LIMIT)
+		downloadLimit = DEFAULT_ZIP_DOWNLOAD_SIZE_LIMIT
 	}
-	limit := uint64(1024 * 1024 * 1024 * 5)
+	limit := uint64(1024 * 1024 * 1024 * downloadLimit)
 	if size >= limit {
-		err := fmt.Errorf("HandleGetPresignedURLMultiObj: Request entity is larger than %v GB, current file size is: %d, and current file count is: %d", float64(limit)/(1024*1024*1024), size, fileCount)
+		err := fmt.Errorf("request entity is larger than %v GB, current prefix size is: %v GB", float64(limit)/(1024*1024*1024), float64(size)/(1024*1024*1024))
```

Comment:
why do the extra math of calculating float64(limit)/(1024*1024*1024), when you already have the downloadLimit value?